File: src/main/java/com/restApi/bankApp/business/abstracts/TransactionService.java
package com.restApi.bankApp.business.abstracts;

import com.restApi.bankApp.entities.Transaction;
import java.util.List;

public interface TransactionService {
    Transaction transferMoney(int fromAccountId, int toAccountId, double amount, String description);
    List<Transaction> getAccountTransactions(int accountId);
    Transaction getTransactionById(Long transactionId);
} 

File: bank-app-frontend/src/app/core/services/transaction.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Transaction } from '../../shared/models/transaction.model';

@Injectable({
  providedIn: 'root'
})
export class TransactionService {
  constructor(private http: HttpClient) {}

  getRecentTransactions(accountId: number): Observable<Transaction[]> {
    return this.http.get<Transaction[]>(`${environment.apiUrl}/transactions/account/${accountId}`);
  }

  createTransaction(transaction: {
    fromAccountId: number;
    toAccountId: number;
    amount: number;
    description: string;
  }): Observable<Transaction> {
    return this.http.post<Transaction>(`${environment.apiUrl}/transactions/transfer`, transaction);
  }

  getTransactionDetails(transactionId: number): Observable<Transaction> {
    return this.http.get<Transaction>(`${environment.apiUrl}/transactions/${transactionId}`);
  }
} 